libs/ledger-live-common/src/platform/logic.ts

```diff
@@ -1,4 +1,281 @@
+  accountId: string,
+  signedTransaction: RawPlatformSignedTransaction,
+  uiNavigation: (
+    account: AccountLike,
+    parentAccount: Account | null,
+    signedOperation: SignedOperation
+  ) => Promise<string>
+): Promise<string> {
+  if (!signedTransaction) {
+    tracking.platformBroadcastFail(manifest);
+    return Promise.reject(new Error("Transaction required"));
+  }
+
+  const account = accounts.find((account) => account.id === accountId);
+  if (!account) {
+    tracking.platformBroadcastFail(manifest);
+    return Promise.reject(new Error("Account required"));
+  }
+
+  const parentAccount = getParentAccount(account, accounts);
+
+  const signedOperation = deserializePlatformSignedTransaction(
+    signedTransaction,
+    accountId
+  );
+
+  return uiNavigation(account, parentAccount, signedOperation);
+}
+
+export type CompleteExchangeRequest = {
+  provider: string;
+  fromAccountId: string;
+  toAccountId: string;
+  transaction: RawPlatformTransaction;
+  binaryPayload: string;
+  signature: string;
+  feesStrategy: string;
+  exchangeType: number;
+};
+export type CompleteExchangeUiRequest = {
+  provider: string;
+  exchange: {
+    fromAccount: AccountLike;
+    fromParentAccount: Account | null;
+    toAccount?: AccountLike;
+    toParentAccount: Account | null;
+  };
+  transaction: TransactionCommon;
+  binaryPayload: string;
+  signature: string;
+  feesStrategy: string;
+  exchangeType: number;
+};
+export function completeExchangeLogic(
+  { manifest, accounts, tracking }: WebPlatformContext,
+  {
+    provider,
+    fromAccountId,
+    toAccountId,
+    transaction,
+    binaryPayload,
+    signature,
+    feesStrategy,
+    exchangeType,
+  }: CompleteExchangeRequest,
+  uiNavigation: (request: CompleteExchangeUiRequest) => Promise<Operation>
+): Promise<Operation> {
+  tracking.platformCompleteExchangeRequested(manifest);
+
+  // Nb get a hold of the actual accounts, and parent accounts
+  const fromAccount = accounts.find((a) => a.id === fromAccountId);
+
+  const toAccount = accounts.find((a) => a.id === toAccountId);
+
+  if (!fromAccount) {
+    return Promise.reject();
+  }
+
+  if (exchangeType === 0x00 && !toAccount) {
+    // if we do a swap, a destination account must be provided
+    return Promise.reject();
+  }
+
+  const fromParentAccount = getParentAccount(fromAccount, accounts);
+  const toParentAccount = toAccount
+    ? getParentAccount(toAccount, accounts)
+    : null;
+  const exchange = {
+    fromAccount,
+    fromParentAccount,
+    toAccount,
+    toParentAccount,
+  };
+
+  const accountBridge = getAccountBridge(fromAccount, fromParentAccount);
+  const mainFromAccount = getMainAccount(fromAccount, fromParentAccount);
+
+  // TODO-STP: fix this
+  // transaction.family = mainFromAccount.currency.family;
+
+  const platformTransaction = deserializePlatformTransaction(transaction);
+
+  // TODO-STP: fix this
+  // platformTransaction.feesStrategy = feesStrategy;
```

Comment:
I need some help to decide what to do with this type error.

Reply:
To get rid of the type error, we can just move this logic bellow, just after the declaration of `processedTransaction` and actually update the `processedTransaction` `feesStrategy `:
```ts
  let processedTransaction = accountBridge.createTransaction(mainFromAccount);

  processedTransaction.feesStrategy = feesStrategy;

  processedTransaction = accountBridge.updateTransaction(
    processedTransaction,
    platformTransaction
  );
```

Also, since not all crypto avec a `feesStrategy` attribute (today, only the Bitcoin and Ethereum families have one, cf. [here](https://github.com/LedgerHQ/ledger-live/search?q=feesStrategy+path%3Alibs%2Fledger-live-common%2Fsrc%2Ffamilies)), we could probably add a condition to this update such as
```ts
  if (feesStrategy && processedTransaction.feesStrategy) {
    processedTransaction.feesStrategy = feesStrategy;
  }
```

Reply:
It is strange as all `TransactionCommon` (LL type) has a `feeStrategy` property 🤔

Reply:
It seems that the `MODAL_PLATFORM_EXCHANGE_COMPLETE` modal does not need the `feeStrategy` info (as it should be inside the transaction object I suppose).
Do you think we can get rid of it?